### PR TITLE
bugfix: fix can't refresh table meta data for oracle

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
fix can not refresh table meta data for oracle

### Ⅱ. Does this pull request fix one issue?
#1719 #1714

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java [hunk 1]
```diff
@@ -94,7 +97,13 @@ private void init(DataSource dataSource, String resourceGroupId) {
             tableMetaExcutor.scheduleAtFixedRate(new Runnable() {
                 @Override
                 public void run() {
-                    TableMetaCache.refresh(DataSourceProxy.this);
+                    if (DataSourceProxy.this.getDbType().equalsIgnoreCase(JdbcConstants.MYSQL)) {
+                        TableMetaCache.refresh(DataSourceProxy.this);
+                    } else if (DataSourceProxy.this.getDbType().equalsIgnoreCase(JdbcConstants.ORACLE)) {
+                        TableMetaCacheOracle.refresh(DataSourceProxy.this);
+                    } else {
+                        logger.error("refresh table meta failed, {} dost not support.", DataSourceProxy.this.getDbType());
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java]: dose

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java [hunk 2]
```diff
@@ -83,6 +83,29 @@ public static TableMeta getTableMeta(final DataSourceProxy dataSourceProxy, fina
...
+            String key = getCacheKey(dataSourceProxy, entry.getValue().getTableName());
+            if (entry.getKey().equals(key)) {
+                try {
+                    TableMeta tableMeta = fetchSchema(dataSourceProxy, entry.getValue().getTableName());
+                    if (!tableMeta.equals(entry.getValue())) {
+                        TABLE_META_CACHE.put(entry.getKey(), tableMeta);
+                        logger.info("table meta change was found, update table meta cache automatically.");
+                    }
+                } catch (SQLException e) {
+                    logger.error("get table meta error:{}", e.getMessage(), e);
```
Comment (xingfudeshi) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java]: By the way,name of logger should be full capitalized,please correct it as well.

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java (hunk 1 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java]: thank you for correct my mistake

**Comment by jsbxyyx** [comment]
please remove TableMetaCache#resultSetMetaToSchema(java.sql.ResultSet rs2, AbstractConnectionProxy conn, String tableName)

the method is never used.

**Comment by slievrly** [comment]
It is recommended to abstract the public methods of TableMetaCache and TableMetaCacheOracle to a base class BaseTableMetaCache.

**Comment by l81893521** [comment]
> please remove TableMetaCache#resultSetMetaToSchema(java.sql.ResultSet rs2, AbstractConnectionProxy conn, String tableName)
> 
> the method is never used.

ok

**Pull request merged**